src/App.tsx

```diff
@@ -1,8 +1,58 @@
 import React from 'react';
+import { GoodsList } from './components/GoodsList';
+import { Button } from './components/Button';
 import './App.css';
 
-const App = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+const URL
+  = 'https://mate-academy.github.io/react_dynamic-list-of-goods/goods.json';
 
-export default App;
+interface AppState {
+  goods: Good[];
+}
+
+async function getGoods(url: string): Promise<Good[]> {
+  const response = await fetch(url);
+
+  return response.json();
+}
+
+export class App extends React.Component<{}, AppState> {
+  state = {
+    goods: [],
+  };
+
+  handleLoadAllButtonClick = async () => {
+    const goods = await getGoods(URL);
+
+    this.setState({ goods });
+  };
+
+  handleLoad5ButtonClick = async () => {
+    const goods = await getGoods(URL);
+
+    this.setState({
+      goods: goods.slice(0, 5),
+    });
+  };
+
+  handleLoadRedButtonClick = async () => {
+    const goods = await getGoods(URL);
+
+    this.setState({
+      goods: goods.filter((good: Good) => good.color === 'red'),
+    });
+  };
+
+  render(): JSX.Element {
```

Comment:
But here will be better to remove `: JSX.Element`, because you have Component class and what if you will want to return `JSX.Element[]` or `string` from render?